File: Project_Go_Stack_by_ListNode/stackspeedtest/stackspeedtest.go
package solution

import (
	"fmt"
	"strconv"
	"time"

	stack "./stack"
)

func Main(max int) {
	var stk *stack.Stack

	timeStart := time.Now()

	stk = stack.Constructor()

	for i := 0; i < max; i++ {
		stk.Push(i)
//		fmt.Printf("stk.Push(%d)\n", i)
	}

	fmt.Printf("stk.Push(%d) done.\n", max)
	fmt.Printf("stk.Search(%d) ... %d\n", 0, stk.Search(0))

	for stk.Empty() == false {
/*
		size    := stk.Size()
		peek, _ := stk.Peek()
		pop , _ := stk.Pop()
		fmt.Printf("size = %d, peek = %d, pop = %d\n", size, peek, pop)
*/
		stk.Pop()

	}

	fmt.Printf("stk.Pop(%d) done.\n", 0)
	fmt.Printf("stk.Empty()   ... %s\n", strconv.FormatBool(stk.Empty()))

	timeEnd := time.Now()

	fmt.Printf("Execute time : %.3f [ms]\n\n", timeEnd.Sub(timeStart).Seconds()*1000)
}

File: Project_C_Stack_by_Segment_no_free/lib/stackspeedtest.c
#include <time.h>
#include "../include/stackspeedtest.h"

// #define DEBUG

int stackspeedtest(int max) {
    Stack *s = stackCreate(65536);

    stackspeedtest2(s, max);
    stackspeedtest2(s, max);

    stackDestroy(s);
}

int stackspeedtest2(Stack *s, int max) {
    clock_t time_start = clock();

    for (int i = 1; i <= max; i++) {
        stackPush(s, i);

#ifdef DEBUG
        printf("stackPush(%d)\n", i);
#endif

    }

    printf("stackPush(%d) done.\n", max);
//  printf("stackSearch_from_front(%d) ... %d\n", max, stackSearch_from_front(s, max));
    printf("stackSearch_from_tail(%d) ... %d\n", 1, stackSearch_from_tail(s, 1));

    while (stackEmpty(s) == false) {

#ifdef DEBUG
        int size = stackSize(s);
        int top  = stackTop(s);
        int pop  = stackPop(s);
        printf("size = %d, top = %d, pop = %d\n", size, top, pop);

#else
        stackPop(s);

#endif

    }

    printf("stackPop() done.\n");
    printf("stackEmpty() ... %s\n", stackEmpty(s) != 0 ? "true": "false");
    printf("destroy stack.\n");

    clock_t time_end = clock();
    printf("Execute time : %f [ms]\n\n", (double)(time_end - time_start)*1000/CLOCKS_PER_SEC);

    return 0;
}
